api-inference-community/docker_images/stanza/app/pipelines/token_classification.py

```diff
@@ -36,14 +36,32 @@ def __call__(self, inputs: str) -> List[Dict[str, Any]]:
         """
         doc = self.model(inputs)
 
-        entities = []
-        for entity in doc.entities:
-            entity_dict = {
-                "entity_group": entity.type,
-                "word": entity.text,
-                "start": entity.start_char,
-                "end": entity.end_char,
-                "score": 1.0,
-            }
-            entities.append(entity_dict)
-        return entities
+        if "ner_model_path" in self.model.config.keys():
+
+            entities = []
+            for entity in doc.entities:
+                entity_dict = {
+                    "entity_group": entity.type,
+                    "word": entity.text,
+                    "start": entity.start_char,
+                    "end": entity.end_char,
+                    "score": 1.0,
+                }
+                entities.append(entity_dict)
+            return entities
+
+        else:
+            pos_tags = []
+
+            for sent in doc.sentences:
+                for word in sent.words:
+                    pos_dict = {
+                        "pos_group": word.upos,
```

Comment:
I don' t `pos_group`  will be understood by the widget.

Is it possible to stick to `entity_group` for the name of the group ?

That will keep the pipeline simpler too probably.

```python
use_type = "ner_model_path" in self.model.config.keys()
...

   "entity_group" : entity.type if use_type else entity.upos

```

Seems slightly cleaner to me too (removing the outer `if/else`).

Reply:
ok I'll do `entity_group`
Thanks for the feedback!

Reply:
What you're iterating over is different depending on the key so it can't simplify the code but I'll return the same name & variables

Reply:
in NER I iterate over `doc.entities` (models that have NER have POS tagging too) and in POS tagging I iterate over `.words` instead so I don't know a neat way to make it into one loop